src/Riok.Mapperly/Descriptors/Mappings/ExistingTarget/SourceObjectMemberDelegateExistingTargetMapping.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Riok.Mapperly.Emit.SyntaxFactoryHelper;
+
+namespace Riok.Mapperly.Descriptors.Mappings.ExistingTarget;
+
+/// <summary>
+/// Represents a mapping which works by accessing an instance member on the source object
+/// and then mapping the result to the target with an existing target mapper.
+/// </summary>
+public class SourceObjectMemberDelegateExistingTargetMapping : ExistingTargetMapping
+{
+    private readonly string _memberName;
+    private readonly IExistingTargetMapping _delegateMapping;
```

Comment:
What would the optional form look like? Because its an existing target mapping it currectly creates `MapToSomething(source.MyMember, target)`. It can't create something like `target = source.MyMember`.